cns/api.go

```diff
@@ -54,7 +55,7 @@ type HTTPService interface {
 }
 
 // IPConfigsHandlerFunc
-type IPConfigsHandlerFunc func(context.Context, IPConfigsRequest) (*IPConfigsResponse, error)
+type IPConfigsHandlerFunc func(context.Context, *IPConfigsRequest) (*IPConfigsResponse, error)
```

Comment:
Now that we are not modifying the request anymore, I think we should revert this function back to taking `IPConfigsRequest` as a value and not a pointer.